src/common/provider/ScaledSizeProvider.cpp

```diff
@@ -26,6 +26,30 @@ QSize ScaledSizeProvider::getScaledSize(const QSize &size)
 	return { static_cast<int>(size.width() * getXScaleFactor()), static_cast<int>(size.height() * getYScaleFactor()) };
 }
 
+QRect ScaledSizeProvider::getScaledRect(const QRect &rect) {
+    return getScaledRect(rect, getXScaleFactor(), getYScaleFactor());
+}
```

Comment:
Two Methods might have been enough here, something like
```
QRect ScaledSizeProvider::getScaledRect(const QRect &rect) {
    auto xScaleFactor = getXScaleFactor();
    auto yScaleFactor = getYScaleFactor();
    return QRect(rect.left() * xScaleFactor, rect.top() * yScaleFactor, rect.width() * xScaleFactor, rect.height() * yScaleFactor);
}
```

Same for unscaled, something like that.

Reply:
but then i will break **kImageAnnotator/example** or potentially other code based on kImageAnnotator because image `devicePixelRatio` not always equals to screen pixelRatio, in kImageAnnotator/example image have 1.0 devicePixelRatio but screen pixelRatio is 2.0

Reply:
Maybe this is a Qt bug? Those two values should be the same. How many monitors do you have? If you have more the one, do you have different scaling on those monitors?

Reply:
I have one monitor, and i don't think it's a Qt bug. 
`qreal QPixmap::devicePixelRatio() const`[`⇢
`](https://doc.qt.io/qt-5/qpixmap.html#devicePixelRatio) may differ from the screen pixelRatio
 you can even set any pixel ratio you want with:
`void QPixmap::setDevicePixelRatio(qreal scaleFactor)`[`⇢
`](https://doc.qt.io/qt-5/qpixmap.html#setDevicePixelRatio)

Reply:
Yes, you can, but we don't set it, it's set by Qt based on the screen scale. The methods  `getXScaleFactor()` and `getYScaleFactor()` use `desktopWidget->logicalDpiX()` and `desktopWidget->logicalDpiY()` to get the dpi and we divide it by the reference dpi count, which is 96.0. I've tested this on Windows, OpenSUSE and Ubuntu and it always returned the correct scale factor. Actually, the icons in the kImageAnnoator are all scaled by those methods, if those wouldn't work, the Icons in the example app would be really small, but they're probably correctly scaled on your machine. But if you have tested it and it doesn't work on you machine, we need to do it differently. Try change the method names to "getScalePixmapRect" and "getUnscaledPixmapRect", then pass only the pixmap and return the rect which is appropriately scaled. As soon as I find some time I will test it against my machines and see if we have a different behavior here, if yes, we will need to investigate this further.